### DISCUSSION TITLE
Packaging Review

### DISCUSSION THREAD

**Discussion by JotaFan** [vliz-be-opsci/pykg2tbl#6] [General]
**Project**
1. Requirements files should have the .in and the .txt (where the .txt should be build using pip-compile or other tools from the .in). 
This would make a stable versioning of the requirements, and those would be the ones in setup.py
2. Add files created by make install in gitignore (/dist)
3. Add linting packages black and isort (at least) and check them in github actions
4. Add pytest-cov to undertand what the file coverage in tests, could just be added as a gh action.
5. Rename maste branch to main

**Code**
1. Do not make * imports
2. Comments above line, not inline
3. Use with when opening files

**Contibuiting**
1. Define rules on PR and Commit (naming, messaging and rules to aproval)
2. Define versioning
3. Add CONTRIBUTING.md

Edit:

Add how to use example on README

**Comment by marc-portier** [comment]
Some point - for - point matching reactions

**Project**
1. Requirements files... --> rather let us introduce poetry
2. Fix .gitignore (/dist)  --> for sure!
3. Add linting ...  --> makes sense but should be 2 distinct steps: `make check` should do flake8 linting, applying automatic fixes (black) is under developer control
4. Add pytest-cov --> cool, go! (then plan to visit our other projects as well)
5. Rename maste branch to main --> cool


**Code**
1. Do not make * imports  --> should be reported by flake8
2. Comments above line, not inline --> same
3. Use with when opening files --> more nuanced (we should at least allow but document and discuss possible motivated exceptions), but in general we should learn to use with context structure more (not only for files)


**Contibuiting**
1. Define rules on PR and Commit (naming, messaging and rules to aproval)
2. Define versioning --> should be release management + also cover pushing up to PyPi and the like (docs to readthedocs)

+ considering all of these in a more general policy for all our py-repos makes sense --> could be a github-policy.md document in the main website.
